dsdl_compiler/libcanard_dsdl_compiler/code_type_template.tmpl

```diff
@@ -115,100 +137,180 @@ uint32_t ${type_name}Encode(${type_name} *source, void *msg_buf, uint32_t offset
   * @param dyn_arr_buf: NULL or Pointer to memory storage to be used for dynamic arrays
   *                     ${type_name} dyn memory will point to dyn_arr_buf memory.
   *                     NULL will ignore dynamic arrays decoding.
+  * @retval offset or ERROR value if < 0
+  */
+int32_t ${type_name}Decode(const CanardRxTransfer* transfer, uint16_t payload_len, ${type_name} *dest, uint8_t **dyn_arr_buf)
+{
+    for (uint32_t c = 0; c < sizeof(${type_name}); c++)
```

Comment:
Why not use memset here? It is available in the standard library, might as well use it.

Reply:
Well, it is banned for me. Memset_s would be, but it might get tricky to get in all environments. So I would rather take the easier solution.

Reply:
How is your solution different from memset? It does not do any of the checks that memset provides.

Reply:
It is not. It just does not raise SDL static code analysis. 
I can add CANARD_ASSERT to check dest pointer is not NULL if it helps.